application/src/scouter/CoralButton.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useEffect, useState } from "react";
+import { AllSushis, ShusiToBeChanged, Sushi, ValuesToBePassed } from "./input-types/AutonomousMapInput";
+const CoralButton:React.FC<ValuesToBePassed> = (props)=>{
+    let storage = props.storage.get()
+    let storageSushis = [props.storage.get()?.Sushi1, props.storage.get()?.Sushi2,props.storage.get()?.Sushi3]
+    const basicSushi: Sushi = {HasHarvested:false, HasSeeded:false}
+    let initialState: boolean = true
+    const defineCoral = ()=>{
+        switch(props.sushiToBeChanged){
+            case ShusiToBeChanged.SUSHI1:{
+                return storageSushis[0]?.HasSeeded ? true: false
+            }
+            case ShusiToBeChanged.SUSHI2:{
+                return storageSushis[1]?.HasSeeded ? true: false
+            }
+            case ShusiToBeChanged.SUSHI3:{
+                return storageSushis[2]?.HasSeeded ? true: false
+            }
+        }
+    }
+
+    const defineCoralText = ()=>{
+        if(defineCoral() == true){
+            return "Seeded"
+        }
+        else{
+            return "Not Seeded"
+        }
+    }
+
+    const defineCoralColor = ()=>{
+        if(defineCoral() == true){
+            return "#22e025"
+        }
+        else{
+            return "#db1616"
+        }
+    }
+
+    const [hasPiled, updateHasPiled] = useState(defineCoral())
+    const[text, updateText] = useState(defineCoralText())
+    const [color, changeColor] = useState(defineCoralColor())
+
+    console.log(hasPiled)
+
+    const setStorage = ()=>{
+        props.storage.set({Sushi1:storageSushis[0]||basicSushi, Sushi2:storageSushis[1]||basicSushi, Sushi3:storageSushis[2]||basicSushi})
+    }
+
+    const changeSushiValueWhenUnseeded = ()=>{
+        storage = props.storage.get()
+        storageSushis = [props.storage.get()?.Sushi1, props.storage.get()?.Sushi2,props.storage.get()?.Sushi3]
+        switch(props.sushiToBeChanged){
+            case ShusiToBeChanged.SUSHI1:{
+                storageSushis[0] = {HasHarvested: !!storageSushis[0]?.HasHarvested, HasSeeded: false}
+            break;
+            }
+            case ShusiToBeChanged.SUSHI2:{
+                storageSushis[1] = {HasHarvested: !!storageSushis[1]?.HasHarvested, HasSeeded: false}
+                break;
+            }
+            case ShusiToBeChanged.SUSHI3:{
+                storageSushis[2] = {HasHarvested: !!storageSushis[2]?.HasHarvested, HasSeeded: false}
+                break;
+            }
+        }
```

Comment:
```suggestion
```suggestion
    const changeSushiValue = (newValue: boolean)=>{
        storage[this.props.sushiToBeChanged].HasSeeded = newValue;
        setStorage();
        }
```